### PR TITLE
Refine and extend the guide/introduction.md.

### PR DESCRIPTION
**Pull request by varsill**
This PR implements the changes described in: https://membraneframework.atlassian.net/browse/MS-261

### CODE REVIEW

File: guide/introduction.md [hunk 1]
```diff
@@ -1,5 +1,36 @@
-# Membrane Framework
+# Membrane Framework guide
+Hello and welcome to the general guide to the Membrane Framework!
```
Comment (DominikWolek) [inline guide/introduction.md:2]:
```suggestion
Hello, and welcome to the general guide to the Membrane Framework!
```

File: guide/introduction.md [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
 
-Hello and welcome to the general guide to the Membrane Framework! If you see something requiring improvement in this guide, feel free to create an issue or open a PR in [this repository](https://github.com/membraneframework/guide). For more information about the framework and more tutorials, check our website: [membrane.stream](https://membrane.stream).
+Membrane is a multimedia processing framework that focuses on reliability, concurrency, and scalability. It is primarily written in Elixir, while some platform-specific or time-constrained parts are written in C. With a range of existing packages and an easy-to-use interface for writing your own, Membrane can be used to process almost any type of multimedia.
```
Comment (DominikWolek) [inline guide/introduction.md:4]:
```suggestion
Membrane is a multimedia processing framework that focuses on reliability, concurrency, and scalability. It is primarily written in Elixir, while some platform-specific or time-constrained parts written in C. With a range of existing packages and an easy-to-use interface for writing your own, Membrane can be used to process almost any type of multimedia.
```

File: guide/introduction.md [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
 
+## Structure of the Membrane Framework
+The Membrane Framework allows you to define your own processing units, called "elements." An element might be able to mux incoming audio and video streams into a container, or play raw audio using your sound card.
 
+Elements can be organized into a "pipeline," which is a sequence of linked elements that performs a specific task. For example, a pipeline might receive an incoming RTSP stream from a webcam and convert it to an HLS stream, or act as a selective forwarding unit (SFU) to implement your own videoconferencing room.
```
Comment (DominikWolek) [inline guide/introduction.md:9]:
```suggestion
Elements can be organized into a "pipeline," - a sequence of linked elements that performs a specific task. For example, a pipeline might receive an incoming RTSP stream from a webcam and convert it to an HLS stream, or act as a selective forwarding unit (SFU) to implement your own videoconferencing room.
```

File: guide/introduction.md [hunk 4]
```diff
@@ -1,5 +1,36 @@
...
+
+### Membrane Core
+[Membrane Core](https://github.com/membraneframework/membrane_core) is the heart and soul of the Membrane Framework. It is written entirely in Elixir and provides the internal mechanisms and developer's interfaces that allow you to prepare processing elements and link them together in a convenient yet reliable way. Note that Membrane Core does not contain any multimedia-specific logic. 
+The documentation for the developer's API is available at [hexdocs](https://hexdocs.pm/membrane_core/readme.html).
+
+### Membrane packages
+Membrane packages provide the multimedia processing logic. They are organized by domain, so a package named after a container (e.g. membrane_mp4_plugin) might contain elements for muxing and demuxing that container. A complete list of all the Membrane packages managed by the Membrane team is available here.
```
Comment (DominikWolek) [inline guide/introduction.md:16]: However it is true, I don't like the "might" in that sentence, it's like a user will have to guess if plugin will contain elements useful for him.

**Review by DominikWolek — CHANGES_REQUESTED** [review]

File: guide/introduction.md (hunk 2 above)
Comment (varsill) [reply guide/introduction.md:4]: I think that `are` is crucial: "while some ... parts ARE written in C". Alternatively we can think of writing: "with some ... parts written in C"

File: guide/introduction.md (hunk 4 above)
Comment (varsill) [reply guide/introduction.md]: done

**Review by DominikWolek — APPROVED** [review]

**Pull request merged**